### PR TITLE
pronoun tagging - beach patrols training

### PR DESCRIPTION
**Pull request by sedgestripe**

### CODE REVIEW

File: resources/dicts/patrols/beach/training/any.json [hunk 1]
```diff
@@ -1,556 +1,944 @@
...
+        "chance_of_success": 60,
+        "exp": 20,
+        "success_text": [
             "app1 listens carefully as p_l explains where to find the best prey within the rich waters of the pools.",
             "app1 picks up the instruction quickly and is soon hopping from pool to pool, picking out the tasty morsels within.",
-            null,
-            "By the end of the patrol, s_c is easily fishing fresh-kill out of the pools with not an ounce of hesitation in their movements."
-    ],
-    "fail_text": [
+            "By the end of the patrol, s_c is easily fishing fresh-kill out of the pools with not an ounce of hesitation in {PRONOUN/s_c/poss} movements.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: it seems this got placed in a list index that's supposed to be null?

File: resources/dicts/patrols/beach/training/any.json [hunk 2]
```diff
@@ -1,556 +1,944 @@
...
+            "By the end of the patrol, s_c is easily fishing fresh-kill out of the pools with not an ounce of hesitation in {PRONOUN/s_c/poss} movements."
+        ],
+        "fail_text": [
             "app1 tries to remember the differences and nuances between the creatures within the pools, but fails to retain any of the information.",
             "s_c struggles to take the lesson seriously and eventually the training session ends early with no real progress made.",
-            null,
+            "app1 tries to poke around in the pools for potential food. Ouch! Oops, that was a sea urchin. app1 limps the whole way home.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: seems to have happened here as well

File: resources/dicts/patrols/beach/training/any.json [hunk 3]
```diff
@@ -1,556 +1,944 @@
...
+        "chance_of_success": 50,
+        "exp": 20,
+        "success_text": [
+            "{PRONOUN/r_c/poss/CAP} Clanmates come running at {PRONOUN/r_c/poss} yowls, and r_c is soon freed, shaken but unharmed. The patrol settles down as everyone comforts each other.",
             "p_l digs through the sand, helping r_c clear it away enough for the other cat to scramble free.",
-            "Without letting themselves panic, s_c carefully assesses the collapsed dune trapping their Clanmate. They point out stable footing for the patrol to dig them out from above, and r_c is recovered, scared but okay.",
-            "Creeping in as close as possible, s_c purrs to their Clanmate, keeping them calm as the patrol carefully digs down to them."
-    ],
-    "fail_text": [
+            "Without letting {PRONOUN/r_c/subject} panic, s_c carefully assesses the collapsed dune trapping {PRONOUN/r_c/poss} Clanmate. {PRONOUN/r_c/subject/CAP} {VERB/r_c/point/points} out stable footing for the patrol to dig {PRONOUN/r_c/object} out from above, and r_c is recovered, scared but okay.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: a subject pronoun tag replaced 'themselves', when it should be a self pronoun tag.   the first four pronouns should be for s_c, not r_c

File: resources/dicts/patrols/beach/training/any.json [hunk 4]
```diff
@@ -1,556 +1,944 @@
...
-    "antagonize_text": null,
-    "antagonize_fail_text": null,
-    "history_text": {
-        "scar": "r_c carries a scar from being trapped on patrol.",
-        "reg_death": "r_c tragically died after being trapped on patrol.",
-        "lead_death": "died after being trapped on patrol"
-    }
-},
-{
+            "As s_c rarely has patience for other cats, r_c tells {PRONOUN/r_c/object} harshly that {PRONOUN/r_c/subject} don't have time for some idiot who got stuck, making s_c resentful as {PRONOUN/r_c/subject} {VERB/r_c/dig/digs} {PRONOUN/r_c/object} out of the sand.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: first pronoun tag should be s_c, not r_c, and last 3 pronoun tags should be s_c, not r_c

File: resources/dicts/patrols/beach/training/any.json [hunk 5]
```diff
@@ -1,556 +1,944 @@
...
+            "Hurt and wheezing, r_c hears the patrol panicking as {PRONOUN/r_c/subject} {VERB/r_c/drift/drifts} to StarClan.",
+            "The patrol eventually manages to rescue r_c, but {PRONOUN/r_c/subject} {VERB/r_c/need/needs} to be helped back to camp, battered and injured.",
+            "The patrol can't help s_c in time, and as s_c starts to drift off to StarClan, the eyes of r_c glint malevolently. Was this...? s_c doesn't have a chance to finish {PRONOUN/r_c/poss} last thought.",
+            "Even though s_c made it out, {PRONOUN/r_c/subject} {VERB/r_c/swear/swears} someone pushed {PRONOUN/r_c/object}, and blame r_c. r_c denies it and says {PRONOUN/r_c/subject} should get s_c back to camp to get {PRONOUN/r_c/poss} injury looked at."
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: first three and last pronoun tag should be s_c, not r_c

File: resources/dicts/patrols/beach/training/any.json [hunk 6]
```diff
@@ -1,556 +1,944 @@
...
+            "patrol_to_p_l",
+            "patrol_to_r_c"
+        ],
+        "intro_text": "Trailing behind p_l, the dune {PRONOUN/r_c/subject}{VERB/r_c/'re/'s} crossing shifts beneath r_c's paws. {PRONOUN/r_c/subject/CAP} {VERB/r_c/look/looks} around wildly for safe footing - just as the sand caves in beneath {PRONOUN/r_c/object} and the world goes dark.",
         "decline_text": "Thankfully, they're on the edge of the obstruction, and wriggle their way clear.",
         "chance_of_success": 50,
         "exp": 20,
         "success_text": [
-            "p_l comes running at their yowls, and r_c is soon freed, shaken but unharmed. The cats settle down as they comfort each other.",
+            "p_l comes running at {PRONOUN/r_c/poss} yowls, and r_c is soon freed, shaken but unharmed. The cats settle down as {PRONOUN/r_c/subject} {VERB/r_c/comfort/comforts} each other.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: last two tags don't need to be there, the last 'they' is referring to the patrol as a whole

File: resources/dicts/patrols/beach/training/any.json [hunk 7]
```diff
@@ -1,556 +1,944 @@
...
             "p_l digs through the loose sand, helping r_c clear it away enough for the other cat to scramble free.",
-            "Without letting themselves panic, s_c carefully assesses the collapsed dune trapping their Clanmate. They find stable footing to dig them out from above, and r_c is recovered, scared but okay.",
-            "Creeping in as close as possible, s_c purrs to r_c, keeping them calm as they carefully dig down to them. r_c blinks up at them, holding on to their presence in the scary situation."
+            "Without letting {PRONOUN/r_c/subject} panic, s_c carefully assesses the collapsed dune trapping {PRONOUN/r_c/poss} Clanmate. {PRONOUN/r_c/subject/CAP} {VERB/r_c/find/finds} stable footing to dig {PRONOUN/r_c/object} out from above, and r_c is recovered, scared but okay.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: first pronoun should be self, not subject.  the first four pronouns should be for s_c, not r_c

File: resources/dicts/patrols/beach/training/any.json [hunk 8]
```diff
@@ -1,556 +1,944 @@
...
+            "Creeping in as close as possible, s_c purrs to r_c, keeping {PRONOUN/r_c/object} calm as {PRONOUN/r_c/subject} carefully dig down to {PRONOUN/r_c/object}. r_c blinks up at {PRONOUN/r_c/object}, holding on to {PRONOUN/r_c/poss} presence in the scary situation."
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: second pronoun should be s_c, not r_c.  last two pronouns should be s_c, not r_c

File: resources/dicts/patrols/beach/training/any.json [hunk 9]
```diff
@@ -1,556 +1,944 @@
...
         ],
         "fail_text": [
             "p_l works all day to free r_c and gets nothing else done.",
-            "As s_c rarely has patience for other cats, r_c tells them harshly that they don't have time for some idiot who got stuck, making s_c resentful as they dig themselves out of the sand.",
-            "Hurt and wheezing, r_c hears p_l panicking as they drift to StarClan.",
-            "p_l eventually manages to rescue r_c, but they need to be helped back to camp, battered and injured."
+            "As s_c rarely has patience for other cats, r_c tells {PRONOUN/r_c/object} harshly that {PRONOUN/r_c/subject} don't have time for some idiot who got stuck, making s_c resentful as {PRONOUN/r_c/subject} {VERB/r_c/dig/digs} {PRONOUN/r_c/object} out of the sand.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: first pronoun tag should be s_c, not r_c, and last 3 pronoun tags should be s_c, not r_c

File: resources/dicts/patrols/beach/training/any.json [hunk 10]
```diff
@@ -1,556 +1,944 @@
...
-    }
-},
-{
+            "scar": "r_c carries a scar from being trapped on patrol",
+            "reg_death": "r_c tragically died after being trapped on patrol.",
+            "lead_death": "died after being trapped on patrol"
+        }
+    },
+    {
         "patrol_id": "bch_train_stuck3",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json:195]: this patrol likely has all the same problems as the previous two

File: resources/dicts/patrols/beach/training/any.json [hunk 11]
```diff
@@ -1,556 +1,944 @@
...
+        "decline_text": "Thankfully, they're on the edge of the obstruction, and wriggle their way clear.",
+        "chance_of_success": 40,
+        "exp": 20,
+        "success_text": [
+            "r_c spends the entire day carefully working {PRONOUN/r_c/poss} way free and gets nothing else done, but at least {PRONOUN/r_c/subject}{VERB/r_c/'re/'s} unharmed.",
+            "s_c tries to remain calm, clawing sand away from {PRONOUN/s_c/poss} muzzle so {PRONOUN/s_c/subject} can breathe. As the shock wears off, {PRONOUN/s_c/subject} {VERB/s_c/realize/realizes} the sand seems thinner on one side of {PRONOUN/s_c/poss} body. With a surge of energy, {PRONOUN/s_c/subject} {VERB/s_c/burst/bursts} through the weak point, terrified and caked in damp sand. {PRONOUN/s_c/subject/CAP} may have a new nightmare to take back with {PRONOUN/s_c/object}, but at least {PRONOUN/s_c/subject}{VERB/s_c/'re/'s} alive.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: it replaced a null value again

File: resources/dicts/patrols/beach/training/any.json [hunk 12]
```diff
@@ -1,556 +1,944 @@
...
+            null
+        ],
+        "fail_text": [
+            "r_c tries {PRONOUN/r_c/poss} best, but unfortunately, {PRONOUN/r_c/subject}{VERB/r_c/'re/'s} only able to work {PRONOUN/r_c/poss} head free. {PRONOUN/r_c/subject/CAP} yowl until {PRONOUN/r_c/poss} voice is hoarse and {PRONOUN/r_c/poss} throat is sore, and thankfully, a patrol passes by after a few hours and frees {PRONOUN/r_c/poss} hapless Clanmate.",
+            "Hurt and wheezing, r_c does {PRONOUN/r_c/poss} best to hang on and wait for a patrol, but {PRONOUN/r_c/subject} {VERB/r_c/succumb/succumbs} to {PRONOUN/r_c/poss} injuries and drift to StarClan alone.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json:289]: replaced a null value again

File: resources/dicts/patrols/beach/training/any.json [hunk 13]
```diff
@@ -1,556 +1,944 @@
...
+            "r_c eventually manages to pull {PRONOUN/r_c/object} free, but {PRONOUN/r_c/subject}{VERB/r_c/'re/'s} wheezing and injured. It's all {PRONOUN/r_c/subject} can do to find some beach grass to shelter in through the night out on the shore until a patrol finds {PRONOUN/r_c/object} the next morning and helps {PRONOUN/r_c/object} to the medicine den.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: first subject should be self, this seems to be a repeating problem.  probably need to be checking for 'themselves' before 'them'?

File: resources/dicts/patrols/beach/training/any.json [hunk 14]
```diff
@@ -1,556 +1,944 @@
...
+        "chance_of_success": 40,
+        "exp": 20,
+        "success_text": [
+            "At least this is a good chance to learn the territory! r_c heads off to map out a field of sand dunes {PRONOUN/r_c/subject}{VERB/r_c/'ve/'s} always found confusing.",
             "r_c clambers over a rise in the ground and stumbles across a tiny, perfect waterfall!",
-            null,
-            "There's nothing to worry about! s_c pads confidently through c_n territory, assured in their safety by the strength of c_n borders."
-    ],
-    "fail_text": [
+            "There's nothing to worry about! s_c pads confidently through c_n territory, assured in {PRONOUN/s_c/poss} safety by the strength of c_n borders.",
```
Comment (scribblecrumb) [inline resources/dicts/patrols/beach/training/any.json]: replaced a null value

**Review by scribblecrumb — CHANGES_REQUESTED** [review]
Not an extensive review, but the start of one.  Two repetitive issues were that "themselves" was being treated as "them", which is incorrect.  It also seems like 'null' outcome values were being replaced with actual outcomes, which will screw things up.

**Review by Ryos00 — APPROVED** [review]

**Review by Ryos00 — APPROVED** [review]

**Pull request merged**